cluster-autoscaler/core/scale_down.go

```diff
@@ -1101,11 +1103,38 @@ func (sd *ScaleDown) scheduleDeleteEmptyNodes(emptyNodes []*apiv1.Node, client k
 				metrics.RegisterScaleDown(1, gpu.GetGpuTypeForMetrics(sd.context.CloudProvider.GPULabel(), sd.context.CloudProvider.GetAvailableGPUTypes(), nodeToDelete, nodeGroupForDeletedNode), metrics.Unready)
 			}
 			result = status.NodeDeleteResult{ResultType: status.NodeDeleteOk}
-		}(node, nodeGroup)
+		}(node, nodeGroup, sd.context.DaemonSetEvictionForEmptyNodes)
 	}
 	return deletedNodes, nil
 }
 
+// create eviction object for all DaemonSet pods on the node
+func daemonSetEviction(clusterSnapshot simulator.ClusterSnapshot, nodeToDelete *apiv1.Node, client kube_client.Interface, dsEvictionTimeout time.Duration,
+	recorder kube_record.EventRecorder, maxGracefulTerminationSec int) {
+	nodeInfo, _ := clusterSnapshot.NodeInfos().Get(nodeToDelete.Name)
+	_, daemonSetPods, _, _ := simulator.FastGetPodsToMove(nodeInfo, true, true, []*policyv1.PodDisruptionBudget{}, time.Now())
+
+	dsEviction := make(chan status.PodEvictionResult, len(daemonSetPods))
+
+	// Perform eviction of DaemonSet pods. We don't want to raise an error if daemonsetPod wasn't evict properly
+	waitForDsEviction := time.Now().Add(dsEvictionTimeout)
+	for _, daemonSetPod := range daemonSetPods {
+		go func(podToEvict *apiv1.Pod) {
+			dsEviction <- evictPod(podToEvict, true, client, recorder, maxGracefulTerminationSec, waitForDsEviction, 5*time.Second)
+		}(daemonSetPod)
+	}
+	// wait for creating eviction of DaemonSet pods
+	for range daemonSetPods {
+		select {
+		case <-dsEviction:
+		case <-time.After(dsEvictionTimeout + 5*time.Second):
+			klog.V(1).Infof("Failed to create DaemonSet eviction for %v seconds on the %v", dsEvictionTimeout, nodeToDelete.Name)
+			return
+		}
+	}
+	klog.V(1).Infof("all DaemonSet pods created eviction on the %v", nodeToDelete.Name)
```

Comment:
I'd remove this line - we'll log something if something goes wrong anyway, so this line doesn't carry much additional information. It does, however, have potential to be spammy, if a lot of empty nodes are deleted in parallel.